modules/search/search.go

```diff
@@ -0,0 +1,124 @@
+package search
+
+import (
+	"github.com/bwmarrin/discordgo"
+	"github.com/lordralex/absol/api"
+	"github.com/lordralex/absol/api/database"
+	"github.com/lordralex/absol/api/logger"
+	"github.com/lordralex/absol/modules/factoids"
+	"github.com/spf13/viper"
+	"strconv"
+	"strings"
+)
+
+type Module struct {
+	api.Module
+}
+
+// Load absol commands API
+func (*Module) Load(ds *discordgo.Session) {
+	api.RegisterCommand("search", RunCommand)
+
+	api.RegisterIntentNeed(discordgo.IntentsGuildMessages, discordgo.IntentsDirectMessages)
+}
+
+func RunCommand(ds *discordgo.Session, mc *discordgo.MessageCreate, _ string, args []string) {
+	if len(args) == 0 {
+		_ = factoids.SendWithSelfDelete(ds, mc.ChannelID, "You must specify a search string!")
+		return
+	} else if len(strings.Join(args, "")) < 3 {
+		_ = factoids.SendWithSelfDelete(ds, mc.ChannelID, "Your search is too short!")
+		return
+	}
+
+	if mc.GuildID != "" {
```

Comment:
This check should be first. You should not trigger the help unless you're in a DM.